### PR TITLE
zqd: Add index post endpoint

### PR DESCRIPTION
**Pull request by mattnibs**
Part of #882

### CODE REVIEW

**Comment by mccanne** [comment]
What about some tests and zapi client-side support?  The ztest stuff you set up is really easy and it would help understand the addition and play around with it.   We should work on zql syntax for index searches and having it all wire-able with zapi will help internalize it all.

**Comment by mattnibs** [comment]
> What about some tests and zapi client-side support? The ztest stuff you set up is really easy and it would help understand the addition and play around with it. We should work on zql syntax for index searches and having it all wire-able with zapi will help internalize it all.

Definitely. I am working on a follow up issue to this that will include zapi commands for index find and create.

File: zqd/handlers.go [hunk 1]
```diff
@@ -473,6 +474,34 @@ loop:
...
+
+	store, ok := s.Storage().(*archivestore.Storage)
+	if !ok {
+		respondError(c, w, r, zqe.E(zqe.Invalid, "space storage does not support creating indexes"))
+		return
+	}
+	if err := store.IndexCreate(ctx, req); err != nil {
+		respondError(c, w, r, err)
+		return
+	}
```
Comment (alfred-landrum) [inline zqd/handlers.go:502]:
```suggestion
	}
	w.WriteHeader(http.StatusNoContent)
```

**Review by alfred-landrum — APPROVED** [review]

File: zqd/storage/archivestore/archivestore.go [hunk 2]
```diff
@@ -101,6 +104,34 @@ func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Rea
 	return archive.Import(ctx, s.ark, zctx, zr)
 }
 
+func (s *Storage) IndexCreate(ctx context.Context, req api.IndexPostRequest) error {
+	var rules []archive.Rule
+	if req.AST != nil {
+		proc, err := ast.UnpackMap(nil, req.AST)
```
Comment (mattnibs) [inline zqd/storage/archivestore/archivestore.go:110]:
```suggestion
		proc, err := ast.UnpackJSON(nil, req.AST)
```

**Pull request merged**